cli/cmd/tapRunner.go

```diff
@@ -73,6 +73,14 @@ func RunMizuTap() {
 
 	targetNamespaces := getNamespaces(kubernetesProvider)
 
+	if config.Config.IsNsRestrictedMode() {
+		if len(targetNamespaces) != 1 || !mizu.Contains(targetNamespaces, config.Config.MizuResourcesNamespace) {
```

Comment:
why contains? you reach this case only when len(targetNamespaces) == 1 

```suggestion
		if len(targetNamespaces) != 1 || targetNamespaces[0] != config.Config.MizuResourcesNamespace {
```